src/Simplex_tree/include/gudhi/Simplex_tree.h

```diff
@@ -2215,13 +2280,18 @@ class Simplex_tree {
     return p;
   };
 
+  //TODO: externalize this method and `decode_extended_filtration`
   /** \brief Extend filtration for computing extended persistence. 
    * This function only uses the filtration values at the 0-dimensional simplices, 
    * and computes the extended persistence diagram induced by the lower-star filtration 
    * computed with these values. 
    * \post Note that after calling this function, the filtration 
    * values are actually modified. The function `decode_extended_filtration()` 
    * retrieves the original values and outputs the extended simplex type.
+   *
+   * @warning Currently only works for @ref SimplexTreeOptions::Filtration_value which are
+   * float types like `float` or `double`, or **signed** integers.
```

Comment:
I thought we just said it does not work for integers, because it expects to scale [min,max] to [-2,-1]?